### PR TITLE
Gamesettings

### PR DESCRIPTION
**Pull request by efodiks**
Added mouse sensitivity and player speed sliders. Waiting for suggestions

### CODE REVIEW

**Comment by efodiks** [comment]
In game
image

**Comment by efodiks** [comment]
In menu
image

File: Assets/Scripts/Common/Constants.cs [hunk 1]
```diff
@@ -2,12 +2,17 @@
 {
   public static class Constants
   {
+    public const float c_defaultMovementSpeed = 12f;
```
Comment (arkadR) [inline Assets/Scripts/Common/Constants.cs:5]: why not private?

File: Assets/Scripts/Common/Constants.cs [hunk 2]
```diff
@@ -2,12 +2,17 @@
...
+    public const float c_defaultMouseSensitivity = 100f;
```
Comment (arkadR) [inline Assets/Scripts/Common/Constants.cs:6]: for public consts (especially in Constants class) I would skip the c_ prefix. It should be used for private consts

File: Assets/Scripts/MouseLook.cs [hunk 3]
```diff
@@ -1,26 +1,31 @@
...
 
     // Update is called once per frame
     void Update()
     {
       if (GameService.Instance.IsPaused)
         return;
 
-      var mouseX = Input.GetAxis("Mouse X") * mouseSensitivity * Time.deltaTime;
-      var mouseY = Input.GetAxis("Mouse Y") * mouseSensitivity * Time.deltaTime;
+      var mouseX = Input.GetAxis("Mouse X") * _settingsManager.MouseSensitivity * Time.deltaTime;
```
Comment (arkadR) [inline Assets/Scripts/MouseLook.cs:27]: Have you looked into the Input.GetAxis differences on different OSes? You mentioned there was a discrepancy

File: Assets/Scripts/PlayerMovement.cs [hunk 4]
```diff
@@ -53,7 +48,7 @@ void ApplyMovementFollowCamera()
       var z = Input.GetAxis("Vertical");
 
       var offset = (transform.right * x) + (transform.up * y) + (Camera.main.transform.forward * z);
-      controller.Move(offset * speed * Time.deltaTime);
+      controller.Move(offset * (_settingsManager.MovementSpeed * Time.deltaTime));
```
Comment (arkadR) [inline Assets/Scripts/PlayerMovement.cs:51]: the brackets aren't needed

**Comment by arkadR** [comment]
> In menu
> image

I would like to see a label showing the current numeric value, but that is low priority. Otherwise looks nice

**Comment by arkadR** [comment]
Offtopic comment: I have seen a tip to make prefabs for every object in the scene (even if used only once). The reason for this is that when we modify the prefab, only its file changes (and not GameScene.unity). This should solve our merging issues

File: Assets/Scripts/PlayerMovement.cs (hunk 4 above)
Comment (efodiks) [reply Assets/Scripts/PlayerMovement.cs:51]: actually they change the multiplication order and it was recommended by rider for efficiency xd

Comment (arkadR) [reply Assets/Scripts/PlayerMovement.cs:51]: well that is interesting

**Pull request merged**